tests/javascript_tests/test_plottable.js

```diff
@@ -0,0 +1,119 @@
+/**
+ *
+ * @author Jamie Morton, Jose Navas Molina, Andrew Hodges & Yoshiki
+ *         Vazquez-Baeza
+ * @copyright Copyright 2013--, The Emperor Project
+ * @credits Jamie Morton, Jose Navas Molina, Andrew Hodges & Yoshiki
+ *          Vazquez-Baeza
+ * @license BSD
+ * @version 0.9.51-dev
+ * @maintainer Yoshiki Vazquez Baeza
+ * @email yoshiki89@gmail.com
+ * @status Development
+ *
+ */
+
+$(document).ready(function() {
+
+  module("Plottable", {
+
+    setup: function(){
+    },
+
+    teardown: function(){
+    }
+  });
+
+  /**
+   *
+   * Test that the Plottable object is initialized correctly, without optional
+   * arguments.
+   *
+   */
+  test("Test constructor", function() {
+    var plot = new Plottable('foo', ['a', 'b', 'c'], [0.2, 0.3, 0.5, 0.2]);
+
+    equal(plot.name, 'foo', "The name matches!");
+    deepEqual(plot.metadata, ['a', 'b', 'c'], "The metadata match!");
+    deepEqual(plot.coordinates, [0.2, 0.3, 0.5, 0.2], "The coordinates "+
+              "match!");
+
+    // optional arguments get default values
+    equal(plot.idx, -1, "The index value was set correctly!");
+    deepEqual(plot.ci, [], "The confidence intervals were set correctly!");
+  });
+  
+  /**
+   *
+   * Test that the Plottable object is initialized correctly, with optional
+   * arguments.
+   *
+   */
+  test("Test constructor optional+", function() {
+    var plot = new Plottable('foo', ['a', 'b', 'c'], [0.2, 0.3, 0.5, 0.2], 1);
+
+    equal(plot.name, 'foo', "The name matches!");
+    deepEqual(plot.metadata, ['a', 'b', 'c'], "The metadata matches!");
+    deepEqual(plot.coordinates, [0.2, 0.3, 0.5, 0.2], "The coordinates "+
+              "match!");
+
+    // optional arguments get default values
+    equal(plot.idx, 1, "The index value was set correctly!");
+    deepEqual(plot.ci, [], "The confidence intervals were set correctly!");
+
+    plot = new Plottable('foo', ['a', 'b', 'c'], [0.2, 0.3, 0.5, 0.2], 1,
+                         [3, 2, 3, 4]);
+
+    equal(plot.name, 'foo', "The name matches!");
+    deepEqual(plot.metadata, ['a', 'b', 'c'], "The metadata matches!");
+    deepEqual(plot.coordinates, [0.2, 0.3, 0.5, 0.2], "The coordinates "+
+              "match!");
+
+    // optional arguments get default values
```

Comment:
This comment do not correspond to the code, since it is not testing that they get the default values.